lib/dependencies.js

```diff
@@ -288,10 +296,10 @@ module.exports = class Dependencies {
    * @param [newRootDir] {string | undefined}
    * @return {Dependencies};
    */
-  static deserialize(dependencyData, newRootDir) {
+  static deserialize(dependencyData, newRootDir, customFS = fs) {
```

Comment:
Providing this as a default seems strange, given the existing if `input/output` is in the same codebase, this code should only use those APIs.

I worry if we leave a default like this in, it may accidentally fallback to not using `input`. 

Also, I don't see any use-case where an alternative beyond `this.input` can be used

Reply:
I did for the sake of tests, as it tests these independently.